### PR TITLE
smb: add login control access parameters to share resource

### PR DESCRIPTION
**Pull request by phlogistonjohn**
Depends on #57294

The login_control attribute contains a list of objects that will be mapped to samba share access parameters. The restrict_access attribute is a boolean that makes access to the share contingent on being listed in the login_control list, as either a listed user or member of a listed group, for any access level other than 'none' (as that always denies access).

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: src/pybind/mgr/smb/tests/test_validation.py [hunk 1]
```diff
@@ -110,3 +110,23 @@ def test_clean_custom_options():
...
+@pytest.mark.parametrize(
+    "value,ok,err_match",
+    [
+        ("tim", True, ""),
+        ("britons\\arthur", True, ""),
+        ("lance a lot", False, "spaces, tabs, or newlines"),
+        ("tabs\ta\tlot", False, "spaces, tabs, or newlines"),
+        ("bed\nivere", False, "spaces, tabs, or newlines"),
+        ("runawa" + ("y" * 122), True, ""),
+        ("runawa" + ("y" * 123), False, "128"),
```
Comment (anthonyeleven) [inline src/pybind/mgr/smb/tests/test_validation.py:124]: Brave brave Sir Robin

**Review by adk3798 — APPROVED** [review]

**Comment by avanthakkar** [comment]
jenkins test make check arm64

**Review by avanthakkar — APPROVED** [review]
LGTM!

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2024-06-21_12:01:59-orch:cephadm-wip-adk-testing-2024-06-20-1417-distro-default-smithi/

Failures

- 10 `in cluster log` type failures. Nothing new here, still cleaning these up
- 3 failures in mds_upgrade_sequence, known issue
- 1 failure in staggered upgrade with agent, known issue

**Pull request merged**